src/main/java/Parser.java

```diff
@@ -0,0 +1,93 @@
+import java.util.ArrayList;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.io.File;
+
+public class Parser {
+    private String command;
+    private Ui uiDisplay;
+    private Storage fileStorage;
+    public Parser(){
+    }
+
+    public Parser(String command,Ui uiDisplay,Storage fileStorage){
+        this.command = command ;
+        this.uiDisplay = uiDisplay;
+        this.fileStorage = fileStorage;
+    }
+
+    public String getCommandType (String command){
+        String [] s ;
+        s = command.split(" ");
+        return s[0];
+    }
+
+    public String todoTaskCommand(String command, ArrayList<Task> tasks,Ui uiDisplay,File f)throws Exception{
+        String [] s = command.split("todo ");
+        tasks.add(new Task(s[1]));
+        String customeriseTopMes = uiDisplay.getTopTwoLine();
+        String taskMes = "   "+tasks.get(tasks.size()-1).toString();
+        String customeriseBottomMes = uiDisplay.getBottomTwoLine(tasks);
+        fileStorage.writeFile("todo",s[1],f);
+        return uiDisplay.parserOutputMess(customeriseTopMes,taskMes,customeriseBottomMes);
+    }
+
+    public String eventCommand(String command,ArrayList<Task> tasks,Ui uiDisplay,File f) throws Exception{
+        String [] s = command.split("event ");
+        String [] temp = s[1].split("/");
+        String taskName = temp[0];
+        String date = formatDate(temp[1].substring(3,temp[1].length()));
+        String customeriseTopMes = uiDisplay.getTopTwoLine();
+        tasks.add(new Deadline(taskName,date));
+        String taskMes = "     "+tasks.get(tasks.size()-1).toString();
+        String customeriseBottomMes = uiDisplay.getBottomTwoLine(tasks);
+        fileStorage.writeFile("deadline",taskName+date,f);
+        return uiDisplay.parserOutputMess(customeriseTopMes,taskMes,customeriseBottomMes);
+    }
+
+    public String deadlineCommand(String command,ArrayList<Task> tasks,Ui uiDisplay,File f) throws Exception{
+        String [] s = command.split("deadline ");
+        String[] temp = s[1].split("/");
+        String taskName = temp[0];
+        String date = formatDate(temp[1].substring(3,temp[1].length()));
+        String customeriseTopMes = uiDisplay.getTopTwoLine();
+        tasks.add(new Event(taskName, date));
+        String taskMes = "     " + tasks.get(tasks.size()-1).toString();
+        String customeriseBottomMes = uiDisplay.getBottomTwoLine(tasks);
+        fileStorage.writeFile("deadline",taskName+date,f);
+        return uiDisplay.parserOutputMess(customeriseTopMes,taskMes,customeriseBottomMes);
+    }
+
+    public String doneCommand(String command,ArrayList<Task> tasks,Ui uiDisplay){
+        if(tasks.size()>0){
+            String [] temp = command.split(" ");
+            int arrPos = Integer.parseInt(temp[1]);
+            tasks.get(arrPos-1).setDone();
```

Comment:
You could get an ArrayOutOfBoundsException if the integer entered is larger than the number of tasks, or a NumberFormatException if the 'done' command is not followed by a number. Also, this command doesn't update/overwrite the saved file with the 'done' update.